sdk/appconfiguration/azure-data-appconfiguration/README.md

```diff
@@ -167,68 +167,130 @@ ConfigurationAsyncClient client = new ConfigurationClientBuilder()
-- setSetting creates a setting if it doesn't exist or overrides an existing setting.
+The following sections provide several code snippets covering some of the most common configuration service tasks, including:
+- [Create a configuration setting][sample_hello_world]
+- [Retrieve a configuration setting][sample_hello_world]
+- [Update an existing configuration setting][sample_hello_world]
+- [Delete a configuration setting][sample_hello_world]
+- [List configuration settings with multiple keys][sample_list_configuration_settings]
+- [List revisions of multiple configuration settings][sample_read_revision_history]
+- [Set a configuration setting to read only][sample_read_only]
+- [Clear read only from a configuration setting][sample_read_only]
+- [Conditional request a configuration setting][sample_conditional_request]
+
+### Create a configuration client
+Create a configuration client by using `ConfigurationClientBuilder` by passing connection string.
 ```Java
 ConfigurationClient client = new ConfigurationClientBuilder()
         .connectionString(connectionString)
         .buildClient();
-ConfigurationSetting setting = client.setSetting("some_key", "some_label", "some_value");
 ```
 
-### Retrieve a Configuration Setting
+### Create a configuration setting
+
+Create a configuration setting to be stored in the configuration store. There are two ways to store a configuration setting:
+- `addConfigurationSetting` creates a setting only if the setting does not already exist in the store.
+- `setConfigurationSetting` creates a setting if it doesn't exist or overrides an existing setting.
 
-Retrieve a previously stored Configuration Setting by calling getSetting.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.getSetting("some_key", "some_label");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
 ```
 
-### Update an existing Configuration Setting
+### Retrieve a configuration setting
 
-Update an existing Configuration Setting by calling setSetting.
+Retrieve a previously stored configuration setting by calling `getConfigurationSetting`.
 ```Java
-ConfigurationClient client = new ConfigurationClientBuilder()
-        .connectionString(connectionString)
-        .buildClient();
-client.setSetting("some_key", "some_label", "some_value");
-ConfigurationSetting setting = client.setSetting("some_key", "some_label", "new_value");
+ConfigurationSetting setting = client.setConfigurationSetting("some_key", "some_label", "some_value");
+ConfigurationSetting retrievedSetting = client.getConfigurationSetting("some_key", "some_label");
+```
+For conditional request, if you want to include the ETag value of `ConfigurationSetting` in the HTTP request, 
```

Comment:
These implementation details aren't useful to an end user (ie. they don't care if a 304 status is returned. That's an implementation detail.)
Suggestion:

```md
If you want to conditionally fetch a configuration setting, set `ifChanged` to true. When `ifChanged` is null, the
configuration setting is only retrieved if it is different than the given `setting`. This is determined by comparing
the ETag of the `setting` to the one in the service to see if they are the same or not. If the ETags are not the same,
it means the configuration setting is different, and its value is retrieved.
```

Reply:
Same with the others below.